projects/e2fsprogs/fuzz/ext2fs_check_directory_fuzzer.cc

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stddef.h>
+#include <stdint.h>
+#include <unistd.h>
+
+#include "ext2fs/ext2fs.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  static const char* pattern = "/dev/shm/ext2XXXXXX";
+  int fd;
+  char* fname;
+
+  // Write our data to a temp file.
```

Comment:
is a file on the disk really necessary here? see https://github.com/google/fuzzing/blob/master/docs/good-fuzz-target.md#io for more context

if having a file on disk is inevitable, I think it would be better to use a constant name, as otherwise we may accidentally spam the bots with these temporary files

Reply:
I believe we need files on disk due to e2fsprogs being a set of filesystem utilities. Not sure how we would get around that, but I'll make the names constant.